packages/core/src/utils/file-utils.ts

```diff
@@ -78,18 +77,44 @@ export function symlinkJunction(to: string, from: string) {
  * returns the first occurrence of the node_modules, or null of none could be found.
  * @param basePath starting point
  */
-export async function findNodeModules(basePath: string): Promise<string | null> {
+export async function findNodeModulesList(
+  basePath: string
+): Promise<{
+  nodeModulesList: string[];
+  basePath: string;
+}> {
   basePath = path.resolve(basePath);
-  const nodeModules = path.resolve(basePath, 'node_modules');
-  try {
-    await fs.promises.stat(nodeModules);
-    return nodeModules;
-  } catch (e) {
-    const parent = path.dirname(basePath);
-    if (parent === basePath) {
-      return null;
-    } else {
-      return findNodeModules(path.dirname(basePath));
+  const nodeModulesList: string[] = [];
+
+  const dirBfsQueue: string[] = (await fs.promises.readdir(basePath)) ?? [];
+
+  while (dirBfsQueue.length > 0) {
+    const dir = dirBfsQueue.pop();
```

Comment:
To get rid of the "not null assertions" below (`dir && ...`), you can rewrite this while statement:

```ts
let dir;
while (dir = dirs.pop()) {
   // [...]
}
```

Reply:
eslint disallow me to write like that, but I switch to a shortter way to do this instaed of "not null assertions"